java/core/src/main/java/co/worklytics/psoxy/gateway/BulkModeConfigProperty.java

```diff
@@ -7,7 +7,26 @@
 public enum BulkModeConfigProperty implements ConfigService.ConfigProperty {
 
     OUTPUT_BUCKET,
+
+    /**
+     * if provided, this path segment will be prepended to keys of output objects
+     */
     ADDITIONAL_TRANSFORMS,
+
+    /**
+     * if provided, this path segment will be removed from keys of input object to produce
+     * keys of corresponding output objects.
+     *
+     * NOTE: objects with keys that do not start with this segment will still be processed; this is
+     * not any kind of filter.
+     */
+    INPUT_BASE_PATH,
```

Comment:
Maybe as `INPUT_PREFIX` could be a better name? Using path it seems like the complete path, but both input and output are going to be the prefix to use between bucket name and object relative path